cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/ConcurrentStreamWriter.java

```diff
@@ -536,10 +571,28 @@ boolean tryWrite() {
     }
 
     /**
+     * Attempts to refresh the underlying FileWriter.
+     *
+     * @return true if become the writer leader and performed the refresh, false otherwise.
+     */
+    boolean refresh() throws IOException {
+      if (!writerFlag.compareAndSet(false, true)) {
+        return false;
+      }
+
+      try {
+        this.getFileWriter().refresh();
```

Comment:
Use instance of here so that you don't have to make every file writer carries a refresh method